core/templates/dev/head/pages/admin/roles_tab/roles_tab_directive.html

```diff
@@ -0,0 +1,100 @@
+<style>
+  .roles-tab .form-horizontal .control-label {
+    text-align: left;
+  }
+  .roles-tab .view-results-table {
+    margin-top: 20px;
+  }
+</style>
+
+<div class="container-fluid roles-tab">
+  <div class="row">
+    <div class="col-md-8">
+      <md-card class="oppia-page-card oppia-long-text">
+        <form class="form-horizontal" name="viewRoleForm" ng-submit="submitRoleViewForm(viewFormValues)">
+          <legend class="text-center">View User Roles</legend>
+          <div class="form-group">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left" for="viewFormValues.method">
+              Select Method
+            </label>
+            <div class="col-md-8">
+              <select ng-model="viewFormValues.method" class="form-control">
+                  <option value="role">By Role</option>
+                  <option value="username">By Username</option>
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="viewFormValues.method==='role'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left">Select Role</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <select ng-options="roleName as roleString for (roleName, roleString) in VIEWABLE_ROLES" ng-model="viewFormValues.role" class="form-control">
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="viewFormValues.method==='username'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label">Enter Username</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <input type="text" name="username" placeholder="Enter username" ng-model="viewFormValues.username" class="form-control">
+            </div>
+          </div>
+
+          <button type="submit" class="btn btn-success"
+                  ng-disabled="(viewFormValues.method==='role' && !viewFormValues.role) || (viewFormValues.method==='username' && !viewFormValues.username)"
+                  value="view role">
+            View Role
+          </button>
+        </form>
+
+        <div ng-show="resultRolesVisible">
+          <table class="table text-center view-results-table">
+            <thead>
+              <tr>
+                <td><strong>Username</strong></td>
+                <td><strong>Role</strong></td>
+              </tr>
+            </thead>
+            <tbody>
+              <tr ng-repeat="(name, role) in result">
+                  <td>{{name}}</td>
```

Comment:
Yep, it is called interpolating the expression. I checked, and I think it might be related to a bug in angular where they still accept the default interpolators in directives (see the "Known Issues" section on [this page](https://docs.angularjs.org/api/ng/service/$interpolate)).

So, <[]> is correct, and the other one was being accepted by mistake. Thanks for coming across this and pointing it out; it's something I'll try and keep an eye out for in the future! (We try not to use {{...}} for angular stuff because we don't want it to be parsed by the Jinja engine.)